openapi-mock/server/mock_controller.js

```diff
@@ -67,32 +85,43 @@ function registerLogger(app) {
     });
 }
 
-function createEndpoints(openApi_doc, api) {
-    if (api.hasOwnProperty("added_endpoints")) {
-        api.added_endpoints.forEach(function (point) {
-            var endpoint = yaml.safeLoad(fs.readFileSync(point.filePath, 'utf8'));
-            if (!openApi_doc["paths"].hasOwnProperty(point.url)) {
-                LOGGER.debug("Adding custom endpoint %s to %s", point.url, api.name)
-                openApi_doc["paths"][point.url] = endpoint;
-                var yml_format = pretty_yaml.stringify(openApi_doc);
-                utility.writeToFile(api.specification_file, yml_format, true);
-            }
-        });
-    }
+function createEndpoints(openApi_doc, api, file_name) {
+
+    api.added_endpoints.forEach(function (point) {
+        var endpoint = yaml.safeLoad(fs.readFileSync(point.filePath, 'utf8'));
+        if (!openApi_doc["paths"].hasOwnProperty(point.url)) {
+            LOGGER.debug("Adding custom endpoint %s to %s", point.url, api.name)
+            openApi_doc["paths"][point.url] = endpoint;
+        }
+    });
+    var yml_format = pretty_yaml.stringify(openApi_doc);
+    utility.writeToFile(file_name, yml_format, true);
 }
 function createMetadataEndpoint(openApi_doc, api, app) {
     try {
-        app.get(api.baseurl + api.metadata, function (req, res) {
+        app.get(api.baseurl + metadata_endpoint, function (req, res) {
             res.type('text/x-yaml')
             res.status(200)
             res.send(openApi_doc)
         });
-        var endpoint = yaml.safeLoad(fs.readFileSync(__dirname + "/resources/OAuth_template.yaml", 'utf8'));
-        if (!openApi_doc["paths"].hasOwnProperty(api.oauth)) {
-            openApi_doc["paths"][api.oauth] = endpoint;
-            var yml_format = pretty_yaml.stringify(openApi_doc);
-            utility.writeToFile(api.specification_file, yml_format, true);
-        }
+        app.post(api.baseurl + ouath_endpoint, function (req, res) {
+            if (req.body.client_id && req.body.client_secret && req.body.grant_type) {
+                res.type('text/json');
+                res.status(200);
+                res.send({ token: 3333 });
+            }
+            else {
+                var message = "Some of the required parameters are missing";
+                res.status(404);
+                res.send({ error: message });
+            }
+        });
+        // var endpoint = yaml.safeLoad(fs.readFileSync(__dirname + "/resources/OAuth_template.yaml", 'utf8'));

```

Comment:
Do not commit commented code as no one will ever delete it. If you want to keep it as backup, git will do it for you

Reply:
removed commented code